core/commands/install_product_command.rb

```diff
@@ -67,14 +68,15 @@ def init
   # Install product on server
   # param node_name [String] name of the node
   def install_product(name)
-    role_file_path = generate_role_file(name)
-    target_path = "roles/#{name}.json"
-    role_file_path_config = "#{@mdbci_config.path}/#{name}-config.json"
-    target_path_config = "configs/#{name}-config.json"
-    extra_files = [[role_file_path, target_path], [role_file_path_config, target_path_config]]
-    @network_config.add_nodes([name])
-    @machine_configurator.configure(@network_config[name], "#{name}-config.json",
-                                    @ui, extra_files)
+    generate_role_file(name).and_then do |role_file_path|
+      target_path = "roles/#{name}.json"
+      role_file_path_config = "#{@mdbci_config.path}/#{name}-config.json"
+      target_path_config = "configs/#{name}-config.json"
+      extra_files = [[role_file_path, target_path], [role_file_path_config, target_path_config]]
+      @network_config.add_nodes([name])
+      return @machine_configurator.configure(@network_config[name], "#{name}-config.json",
```

Comment:
You do not need to return. The last statement of the block becomes the result of the `and_then` method.